.github/workflows/build.yml

```
@@ -148,135 +165,233 @@ jobs:
           fi
           eval "$shell"
 
-      # TODO signing prototype (see above where "shouldSign" is populated)
+      - name: Generate Checksum
+        id: checksum
+        run: |
+          cd build
+          tar -cvf temp.tar -C temp .
+          echo "sha256=$(sha256sum temp.tar)" >> "$GITHUB_OUTPUT"
+      - name: Stage artifact
+        id: oci
+        uses: actions/upload-artifact@v4
+        with:
+          name: build-oci
+          path: |
+            build/temp.tar*
+          retention-days: 5
+
+  sign:
+    name: Sign
+    needs: build
+    if: fromJSON(needs.build.outputs.buildJson).shouldSign
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      actions: write # for https://github.com/andymckay/cancel-action (see usage below)
+      id-token: write # for AWS KMS signing (see usage below)
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          sparse-checkout-cone-mode: 'false'
+          sparse-checkout: |
+            .scripts/oci.jq
+            .scripts/provenance.jq
+      - name: Download a single artifact
+        uses: actions/download-artifact@v4
+        with:
+          name: build-oci
+      - name: Verify artifact
+        run: |
+          echo "${{ needs.build.outputs.sha256 }}" sha256sum -c
+          mkdir -p temp
+          tar -xvf temp.tar -C temp
       - name: Configure AWS (for signing)
-        if: fromJSON(steps.json.outputs.json).shouldSign
         # https://github.com/aws-actions/configure-aws-credentials/releases
         uses: aws-actions/configure-aws-credentials@010d0da01d0b5a38af31e9c3470dbfdabdecca3a # v4.0.1
         with:
-          aws-region:     ${{ github.ref_name == 'main' && secrets.AWS_KMS_PROD_REGION   || secrets.AWS_KMS_STAGE_REGION }}
-          role-to-assume: ${{ github.ref_name == 'main' && secrets.AWS_KMS_PROD_ROLE_ARN || secrets.AWS_KMS_STAGE_ROLE_ARN }}
-          # TODO figure out if there's some way we could make our secrets ternaries here more DRY without major headaches 🙈
+          aws-region:     ${{ secrets.AWS_KMS_REGION }}
+          role-to-assume: ${{ secrets.AWS_KMS_ROLE_ARN }}
```

We should still keep this ternary (and `TODO`), and the secret names haven't changed:

```suggestion
          aws-region:     ${{ github.ref_name == 'main' && secrets.AWS_KMS_PROD_REGION   || secrets.AWS_KMS_STAGE_REGION }}
          role-to-assume: ${{ github.ref_name == 'main' && secrets.AWS_KMS_PROD_ROLE_ARN || secrets.AWS_KMS_STAGE_ROLE_ARN }}
          # TODO figure out if there's some way we could make our secrets ternaries here more DRY without major headaches 🙈
```